javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -1048,11 +1048,15 @@ public int length() {
 
     @Override
     public Array<T> reverse() {
-        final Object[] arr = new Object[delegate.length];
-        for (int i = 0; i < delegate.length; i++) {
-            arr[delegate.length - 1 - i] = delegate[i];
+        if (size() <= 1) {
+            return this;
+        } else {
+            final Object[] arr = new Object[delegate.length];
+            for (int i = 0; i < delegate.length; i++) {
+                arr[delegate.length - 1 - i] = delegate[i];
+            }
```

Comment:
I think this form of optimization makes most sense for code that is not optimal. But it should not prevent us from writing optimal code if it is still concise and maintainable.

It might also reduce the effort for an escape analysis. JIT: "Already on stack? Ok, nothing to do."